### PR TITLE
implementing listing education user

### PR DESCRIPTION
**Pull request by nirajacharya2**
This PR implements listing of an education user
https://owncloud.dev/libre-graph-api/#/educationUser/ListEducationUsers
https://owncloud.dev/libre-graph-api/#/educationUser/GetEducationUser

### CODE REVIEW

**Review by phil-davis — COMMENTED** [review]
Looks good - for the start we just support listing and getting the existing 4 user attributes, the same ones as we support for the ordinary User API.

We can have unit tests for the EducationUser class that will be very similar to the ones for the User class.

And add unit tests for the new getEducationUsers and getEducationUserById methods. Those will have to mock the actual AI responses etc.

Then we need to work out how to start up an ocis instance with working education user endpoints. For a start, we don't need to be able to write to those endpoints. We can just have a system that already has education users set up, and we can list and get them in integration tests.

File: tests/unit/Owncloud/OcisPhpSdk/EducationUserTest.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * @return void
+     */
+    public function testgetEducationUsers()
+    {
+        $ocis = $this->createMock(Ocis::class);
+        /** @phan-suppress-next-line PhanUndeclaredMethod */
+        $user = $ocis->getEducationUsers();
+        $this->assertContainsOnlyInstancesOf(EducationUser::class, $user);
```
Comment (phil-davis) [inline tests/unit/Owncloud/OcisPhpSdk/EducationUserTest.php:19]: what happens if `$user` is an empty array?
(if something went wrong and there are exactly 0 education users)

Maybe the test will pass.

For the unit tests, we will know exactly how many education users exist, so you can also do `assertCount` to check that the array of `EducationUser` has the expected size.

File: tests/unit/Owncloud/OcisPhpSdk/UserTest.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 use OpenAPI\Client\Model\User;
 use Owncloud\OcisPhpSdk\Exception\InvalidResponseException;
-use Owncloud\OcisPhpSdk\User as SdkUser;
+use Owncloud\OcisPhpSdk\BaseUser as SdkUser;
```
Comment (phil-davis) [inline tests/unit/Owncloud/OcisPhpSdk/UserTest.php:7]: Why do we need to change this?
`class User extends BaseUser`

So I think that the tests will pass without this change?

**Comment by phil-davis** [comment]
`PhanDeprecatedImplicitNullableParam Implicit nullable parameters`
Those were fixed in #248 
Please rebase and get CI green again.

**Review by individual-it — APPROVED** [review]

**Pull request merged**